### PR TITLE
Fix password reset enumeration users

### PR DESCRIPTION
**Pull request by tomekkowalczyk**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ v]
| New feature/enhancement? (use the a.x branch)      | [x ]
| Deprecations?                          | [x ]
| BC breaks? (use the c.x branch)        | [x ]
| Automated tests included?              | [ yes] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:

Risk level:  **MEDIUM**
Impact: 	**Enumerate users**
Vulnerability use conditions: 	**Access to login page**

There are different repsonses when we are trying to reset password in using an existing user, and when the account doesn’t exist.

**Existing user:**
image

image

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2.

### CODE REVIEW

File: app/bundles/UserBundle/Controller/PublicController.php [hunk 1]
```diff
@@ -37,18 +37,16 @@ public function passwordResetAction()
                 $data = $form->getData();
                 $user = $model->getRepository()->findByIdentifier($data['identifier']);
 
-                if (null == $user) {
-                    $form['identifier']->addError(new FormError($this->translator->trans('mautic.user.user.passwordreset.nouserfound', [], 'validators')));
```
Comment (escopecz) [inline app/bundles/UserBundle/Controller/PublicController.php:34]: Please remove the `mautic.user.user.passwordreset.nouserfound` message from the ini file too as you removed all its usages.

**Review by escopecz — CHANGES_REQUESTED** [review]
Thanks for taking care of this issue! I don't quite understand the solution though. If the user exists you get message:

> If this user exist, you have successfully reset your password. You may now login.

If the user does not exist then you get redirect without a message. So the responses are still different whether the user exists or not.

Please write a functional test for those scenarios asserting that the output is the same. That should make it clear. It is also requirement to get the PR merged.

File: app/bundles/UserBundle/Controller/PublicController.php (hunk 1 above)
Comment (tomekkowalczyk) [reply app/bundles/UserBundle/Controller/PublicController.php:34]: oki

**Comment by tomekkowalczyk** [comment]
@escopecz 
`If the user does not exist then you get redirect without a message. So the responses are still different whether the user exists or not.`

The point is that the answers vary depending on whether the user exists or not. This way the attacker can check which email addresses are in the system

**Comment by escopecz** [comment]
> @escopecz `If the user does not exist then you get redirect without a message. So the responses are still different whether the user exists or not.`
> 
> The point is that the answers vary depending on whether the user exists or not. This way the attacker can check which email addresses are in the system

That's the point I'm making. This PR does not solve this problem.

**Comment by tomekkowalczyk** [comment]
@escopecz 

Can you justify why in your opinion it does not solve this problem? We have a unified response, regardless of whether the user exists or not, we receive the same message. Additionally, no redirection to the login page if the user existed in the database. If the PR does not solve the problem, ask for suggestions on how to solve the problem.

**Review by escopecz — DISMISSED** [review]
I was reading the changes poorly. The message will be the same whether the user exists or not and it will always redirect to the login path. All good from code perspective. Thanks! 👍

**Comment by escopecz** [comment]
Let me know if you need some help with the tests.

**Comment by tomekkowalczyk** [comment]
@escopecz Tests have been added

**Review by escopecz — COMMENTED** [review]
Looks great! Please fix the CS check.

**Comment by tomekkowalczyk** [comment]
@escopecz I fixed CSFixer

**Comment by tomekkowalczyk** [comment]
We have build CI error: `There was an error running the uploader: Error uploading to https://codecov.io: Error: There was an error fetching the storage URL during POST: 404 - {'detail': ErrorDetail(string='Unable to locate build via Github Actions API. Please upload with the Codecov repository upload token to resolve issue`

**Comment by escopecz** [comment]
That's a network issue. I'm re-running the failed job. Should be green in 10 minutes

**Review by escopecz — DISMISSED** [review]
The code looks good 👍

**Comment by tomekkowalczyk** [comment]
Hi, can we merge the changes now?

**Comment by escopecz** [comment]
@tomekkowalczyk  It needs 2 things:
1. Resolve the conflict
2. One more approval from any tester.

**Comment by tomekkowalczyk** [comment]
@annamunk You will verify the changes?

**Review by annamunk — APPROVED** [review]
Works well ✔️ Mautic shows the same message for existing and non-existing user.

reset_2
reset_1

**Review by escopecz — APPROVED** [review]
Thank you! Great work! 👍

**Pull request merged**